examples/complex_wallet_examples/create_faucet_server/internal/handlers/topup.go

```diff
@@ -0,0 +1,73 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
+	"github.com/bsv-blockchain/go-wallet-toolbox-faucet-server/internal/methods"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/storage"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wallet"
+	"github.com/gofiber/fiber/v2"
+)
+
+type TopUpRequest struct {
+	Outpoint string `json:"outpoint"` // Format: "txid:outputIndex"
+}
+
+type TopUpResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message,omitempty"`
+}
+
+type TopUpDeps = methods.FaucetDeps
+
+func NewTopUpHandler(deps TopUpDeps) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		var req TopUpRequest
+		if err := c.BodyParser(&req); err != nil || req.Outpoint == "" {
+			return c.Status(http.StatusBadRequest).JSON(TopUpResponse{Status: "error", Message: "outpoint required (format: txid:outputIndex)"})
+		}
+
+		// Parse outpoint into txid and output index
+		parts := strings.Split(req.Outpoint, ":")
+		if len(parts) != 2 {
+			return c.Status(http.StatusBadRequest).JSON(TopUpResponse{Status: "error", Message: "invalid outpoint format, expected txid:outputIndex"})
+		}
+
+		txid := parts[0]
+		outputIndexStr := parts[1]
+		outputIndex, err := strconv.ParseUint(outputIndexStr, 10, 32)
```

Comment:
wouldn't it be simpler to use object with txid and index fields (or simply transaction.Outpoint) as a field in request? 


If not, then maybe use `transaction.OutpointFromString` as this code is already in go-sdk